File: examples/java7-example/src/main/java/nz/bradcampbell/java7example/State.java
package nz.bradcampbell.java7example;

import android.os.Parcel;
import android.os.Parcelable;
import java.util.Date;

import paperparcel.PaperParcel;

@PaperParcel
public final class State implements Parcelable {
  public static Parcelable.Creator<State> CREATOR = PaperParcelState.CREATOR;

  private final int count;

  // Non-private variable can be assigned and read directly; it doesn't need a corresponding
  // constructor argument/setter method and an accessor method.
  Date modificationDate;

  // Excluded from the parcelling/unparcelling process
  private transient long somethingToExclude = 1000L;

  public State(int count) {
    this.count = count;
  }

  public int getCount() {
    return count;
  }

  @Override public int describeContents() {
    return 0;
  }

  @Override public void writeToParcel(Parcel dest, int flags) {
    PaperParcelState.writeToParcel(this, dest, flags);
  }
}

File: examples/kotlin-example/src/main/kotlin/nz/bradcampbell/kotlinexample/State.kt
package nz.bradcampbell.kotlinexample

import paperparcel.PaperParcelable
import paperparcel.PaperParcel
import java.util.Date

@PaperParcel
data class State(
    val count: Int,
    val modificationDate: Date
) : PaperParcelable {
  @Transient val somethingToExclude = 10000L

  companion object {
    @JvmField val CREATOR = PaperParcelState.CREATOR
  }
}
